packages/react/src/components/Select/Select.test.tsx

```diff
@@ -225,9 +239,72 @@ describe('Select', () => {
       expect(() => renderSingleSelect({ options: [] })).not.toThrow();
     });
 
+    it('Renders formatted label if given', () => {
+      const formattedLabelText = 'Dolor sit amet';
+      const formattedOption: SingleSelectOption = {
+        label: 'Lorem ipsum',
+        value: 'test',
+        formattedLabel: <span>{formattedLabelText}</span>,
+      };
+      const options = [formattedOption];
+      renderSingleSelect({ options });
+      expect(screen.getByText(formattedLabelText)).toBeInTheDocument();
+    });
+
+    it('Renders a search field with given label which is empty by default', () => {
+      const searchLabel = 'Søk';
+      renderSingleSelect({ searchLabel });
+      const searchField = screen.getByRole('textbox', { name: searchLabel });
+      expect(searchField).toBeInTheDocument();
+      expect(searchField).toHaveValue('');
+    });
+
+    it('Hides label when user types in the search field', async () => {
+      const selectedOptionIndex = 1;
+      const selectedOption = singleSelectOptions[selectedOptionIndex];
+      renderSingleSelect({ value: selectedOption.value });
+      await act(() => user.type(screen.getByRole('textbox'), 'a'));
+      expect(screen.queryByText(
+        (content, element) => element?.tagName.toLowerCase() === 'span'
+          && content === selectedOption.label
+      )).toBeFalsy();
+    });
+
+    it('Calls optionSearch function when user types in the search field', async () => {
+      renderSingleSelect();
+      await act(() => user.type(screen.getByRole('textbox'), 'a'));
+      expect(optionSearch).toHaveBeenCalledTimes(1);
+      expect(optionSearch).toHaveBeenCalledWith(singleSelectOptions, 'a');
+    });
+
+    it('Sorts options and selects the first when user types in the search field', async () => {
+      renderSingleSelect();
+      await act(() => user.type(screen.getByRole('textbox'), 'a'));
+      expect(screen.getAllByRole('option')[0]).toHaveValue(sortedOptions[0].value);
+      expect(getCombobox()).toHaveValue(sortedOptions[0].value);
+    });
+
+    it('Resets keyword, but keeps sorted list of options, when user starts browsing the list', async () => {
+      renderSingleSelect();
+      await act(() => user.type(screen.getByRole('textbox'), 'a'));
+      await act(() => user.keyboard('{ArrowDown}'));
+      expect(screen.getByRole('textbox')).toHaveValue('');
+      expect(screen.getAllByRole('option')[0]).toHaveValue(sortedOptions[0].value);
+    });
+
+    it('Does not reset keyword while user is writing', async () => {
+      renderSingleSelect()
+      const keyword = 'abc';
+      await act(() => user.type(screen.getByRole('textbox'), keyword));
```

Comment:
Trenger vi act() her? Fungerer det ikke å følge anbefalingene fra RTL, eller får vi samme feil som du nevnte tidligere fortsatt? 🤔 (Tenker på alle steder act er brukt i testene)
```suggestion
      await user.type(screen.getByRole('textbox'), keyword));
```

Reply:
Ser på dette i en separat sak.